src/commands/install.ts

```diff
@@ -143,17 +144,41 @@ function getRollupReplaceKeys(env: EnvVarReplacements): Record<string, string> {
  * field instead of the CJS "main" field.
  */
 function resolveWebDependency(dep: string): DependencyLoc {
-  // if dep includes a file extension, check that dep isn't a package before returning
+  // if dep points directly to a file within a package, return that reference.
+  // No other lookup required.
   if (path.extname(dep) && !validatePackageName(dep).validForNewPackages) {
     const isJSFile = ['.js', '.mjs', '.cjs'].includes(path.extname(dep));
     return {
       type: isJSFile ? 'JS' : 'ASSET',
       loc: require.resolve(dep, {paths: [cwd]}),
     };
   }
+  // If dep is a path within a package (but without an extension), we first need 
+  // to check for an export map in the package.json. If one exists, resolve to it.
+  const packageName = getPackageNameFromSpecifier(dep);
+  if (packageName !== dep) {
+    const [packageManifestLoc, packageManifest] = resolveDependencyManifest(packageName, cwd);
+    if (packageManifestLoc && packageManifest && packageManifest.exports) {
+      const packageEntrypoint = `./${dep.replace(packageName + '/', '')}`;
+      const exportMapEntry = packageManifest.exports[packageEntrypoint];
+      const exportMapValue = exportMapEntry?.browser || exportMapEntry?.import || exportMapEntry?.default || exportMapEntry?.require || exportMapEntry;
```

Comment:
My gut reaction is to want and centralize these entries somewhere, but I realize this is really unique to this one scenario and won’t apply to the Rollup plugins’ concept of entry points.

Reply:
Lets centralize when we need to, right now `resolveWebDependency()` is the only place where we need this so it's nice to have the full logic resolution flow outlined in a central place.